csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/Assets.cs

```diff
@@ -0,0 +1,162 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+
+namespace Semmle.Extraction.CSharp.DependencyFetching
+{
+    /// <summary>
+    /// Class for parsing project.assets.json files.
+    /// </summary>
+    internal class Assets
+    {
+        private readonly ProgressMonitor progressMonitor;
+
+        private static readonly string[] netFrameworks = new[] {
+            "microsoft.aspnetcore.app.ref",
+            "microsoft.netcore.app.ref",
+            "microsoft.netframework.referenceassemblies",
+            "microsoft.windowsdesktop.app.ref",
+            "netstandard.library.ref"
+        };
+
+        internal Assets(ProgressMonitor progressMonitor)
+        {
+            this.progressMonitor = progressMonitor;
+        }
+
+        /// <summary>
+        /// Class needed for deserializing parts of an assets file.
+        /// It holds information about a reference.
+        /// </summary>
+        private class ReferenceInfo
+        {
+            /// <summary>
+            /// This carries the type of the reference.
+            /// We are only interested in package references.
+            /// </summary>
+            public string Type { get; set; } = "";
+
+            /// <summary>
+            /// If not a .NET framework reference we assume that only the files mentioned
+            /// in the compile section are needed for compilation.
+            /// </summary>
+            public Dictionary<string, object> Compile { get; set; } = new();
+        }
```

Comment:
I think you can simplify this class to the below if you're okay with changing `info.Compile.` to `info.Compile?.`.
```C#
private record class ReferenceInfo(string? Type, Dictionary<string, object>? Compile);
```

Reply:
Hah, that is funny: It actually started out by being a record in the early stages of the implementation, but then it was a hazzle (and ugly) that null checks needs to be performed in various locations, which is why it became a class. However, with the current implementation it will be much nicer, if it is turned into a record again 😄